sctp/src/queue/pending_queue.rs

```diff
@@ -25,16 +25,52 @@ impl PendingQueue {
         PendingQueue::default()
     }
 
+    /// Appends a chunk to the back of the pending queue.
     pub(crate) async fn push(&self, c: ChunkPayloadData) {
         self.n_bytes.fetch_add(c.user_data.len(), Ordering::SeqCst);
+        self.queue_len.fetch_add(1, Ordering::SeqCst);
+
         if c.unordered {
             let mut unordered_queue = self.unordered_queue.lock().await;
             unordered_queue.push_back(c);
         } else {
             let mut ordered_queue = self.ordered_queue.lock().await;
             ordered_queue.push_back(c);
         }
-        self.queue_len.fetch_add(1, Ordering::SeqCst);
+    }
+
+    /// Appends chunks to the back of the pending queue.
+    ///
+    /// # Panics
+    ///
+    /// If it's a mix of unordered and ordered chunks.
+    pub(crate) async fn append(&self, chunks: Vec<ChunkPayloadData>) {
```

Comment:
Related to the conversation about `push`, this can be `chunks: impl IntoIterator<ChunkPayloadData>` to allow for more flexibility for the caller. It would require structuring this method a bit differently however